src/main/java/com/basho/riak/client/core/query/timeseries/Cell.java

```diff
@@ -143,89 +197,92 @@ public Cell(Date timestampValue)
      */
     public static Cell newTimestamp(long rawTimestampValue)
     {
-        final RiakTsPB.TsCell tsCell = RiakTsPB.TsCell.newBuilder().setTimestampValue(rawTimestampValue).build();
-        return new Cell(tsCell);
+        final Cell cell = new Cell();
+        cell.initTimestamp(rawTimestampValue);
+        return cell;
     }
 
     public boolean hasVarcharValue()
     {
-        return pbCell.hasVarcharValue();
+        return ((typeBitfield & 0x00000001) == 0x00000001);
     }
 
     public boolean hasLong()
     {
-        return pbCell.hasSint64Value();
+        return ((typeBitfield & 0x00000002) == 0x00000002);
     }
 
-    public boolean hasTimestamp()
+    public boolean hasDouble()
     {
-        return pbCell.hasTimestampValue();
+        return ((typeBitfield & 0x00000004) == 0x00000004);
     }
 
-    public boolean hasBoolean()
+    public boolean hasTimestamp()
     {
-        return pbCell.hasBooleanValue();
+        return ((typeBitfield & 0x00000008) == 0x00000008);
     }
 
-    public boolean hasDouble()
+    public boolean hasBoolean()
     {
-        return pbCell.hasDoubleValue();
+        return ((typeBitfield & 0x00000010) == 0x00000010);
     }
 
     public String getVarcharAsUTF8String()
     {
-        return pbCell.getVarcharValue().toStringUtf8();
+        return varcharValue;
     }
 
     public BinaryValue getVarcharValue()
     {
-        return BinaryValue.unsafeCreate(pbCell.getVarcharValue().toByteArray());
+        return BinaryValue.unsafeCreate(varcharValue.getBytes(CharsetUtils.UTF_8));
     }
 
     public long getLong()
     {
-        return pbCell.getSint64Value();
+        return sint64Value;
     }
 
     public double getDouble()
     {
-        return pbCell.getDoubleValue();
+        return doubleValue;
     }
 
     public long getTimestamp()
     {
-        return pbCell.getTimestampValue();
+        return timestampValue;
     }
 
     public boolean getBoolean()
     {
-        return pbCell.getBooleanValue();
+        return booleanValue;
     }
 
     public RiakTsPB.TsCell getPbCell()
     {
-        return pbCell;
-    }
+        final RiakTsPB.TsCell.Builder builder = RiakTsPB.TsCell.newBuilder();
 
-    public OtpErlangObject getErlangObject() {
-        if (pbCell.hasVarcharValue()) {
-            return new OtpErlangBinary(pbCell.getVarcharValue().toByteArray());
+        if (hasVarcharValue())
+        {
+            builder.setVarcharValue(ByteString.copyFromUtf8(varcharValue));
         }
-        if (pbCell.hasSint64Value()) {
-            return new OtpErlangLong(pbCell.getSint64Value());
+        if (hasLong())
```

Comment:
Should these be `else if` since it's a pretty serious error for more than one value to be set?